doc/fwutil/fwutil.md

```diff
@@ -0,0 +1,439 @@
+                   CPLD        5                   <image_path>/cpld.bin  10                  pending update  Power cycle is required
+                   FPGA        5                   <image_path>/fpga.bin  5                   up to date      Power cycle is required
+New FW will be installed, continue? [y/n]: y
+
+...
+FW update in progress ...
+...
+
+Summary:
+
+Chassis   Module   Component   Status
+--------  -------  ----------  -------
+Chassis1  N/A      BIOS        update successful
+                   CPLD        update failed
+                   FPGA        up to date
+```
+
+2. Modular chassis platform
+```bash
+root@sonic:~# fwutil update --image=next
+Chassis   Module   Component   Version             Image                  Firmware            Status          Info
+--------  -------  ----------  --------            ------                 ---------           -------         -----
+Chassis1           BIOS        0ACLH004_02.02.007  <image_path>/bios.bin  0ACLH004_02.02.010  pending update  Cold reboot is required
+                   CPLD        5                   <image_path>/cpld.bin  10                  pending update  Power cycle is required
+                   FPGA        5                   <image_path>/fpga.bin  5                   up to date      Power cycle is required
+          Module1  ASIC        29.2000.1886        <image_path>/asic.bin  29.2000.1894        pending update  Cold reboot is required
+                   CPLD        5                   <image_path>/cpld.bin  10                  pending update  Power cycle is required
+                   FPGA        5                   <image_path>/fpga.bin  5                   up to date      Power cycle is required
+New FW will be installed, continue? [y/n]: y
+
+...
+FW update in progress ...
+...
+
+Summary:
+
+Chassis   Module   Component   Status
+--------  -------  ----------  -------
+Chassis1           BIOS        update successful
+                   CPLD        update successful
+                   FPGA        up to date
+          Module1  ASIC        update successful
+                   CPLD        update failed
+                   FPGA        up to date
+```
+
+**Supported options:**
+1. -y|--yes - automatic yes to prompts. Assume "yes" as answer to all prompts and run non-interactively
+2. -f|--force - install FW regardless the current version
+3. -i|--image - update FW using current/next SONiC image
+
+**Note:** the default option is _--image=current_
+
+# 3 Flows
+
+## 3.1 Show components status
+
+![Show components status flow](images/show_status_flow.svg "Figure 2: Show components status flow")
+
+###### Figure 2: Show components status flow
+
+## 3.2 Install component FW
+
+### 3.2.1 Non modular chassis platform
+
+![FW install (non modular) flow](images/install_non_modular_flow.svg "Figure 3: FW install (non modular) flow")
```

Comment:
install_component_firmware has been removed. Is itbetter to replace it with chassis.component[index].install_firmware?
Also, a flow which expresses how to map the components' name to their index is required.